tests.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+# Modified version of script from here:
+#   http://stackoverflow.com/questions/26082444/how-to-work-around-travis-cis-4mb-output-limit
+
+NUM_LINES=500
+
+# Helper functions
+dump_output() {
+  echo "Dumping the last $NUM_LINES lines of output:"
+  tail -${NUM_LINES} ${BUILD_OUTPUT}
+}
+
+kill_ping_loop() {
+  kill ${PING_LOOP_PID}
+}
+
+clean_up() {
+  # The build finished without returning an error so dump a tail of the output
+  dump_output
+  # Nicely terminate the ping output loop
+  kill_ping_loop
+}
+
+error_handler() {
+  echo "ERROR: An error was encountered with the build."
+  clean_up
+  exit 1
+}
+
+
+# Abort on error
+set -e
+
+if [[ $# -ne 1 ]]; then
+  echo "ERROR: Wrong # of arguments"
+  exit 1
+fi
+
+if [[ "$1" == "integration" ]]; then
+  # Redirect output to file, ping Travis intermittently so it doesn't kill the test.
+  PING_SLEEP=30s
+  BUILD_OUTPUT=/tmp/build.out
+
+  touch ${BUILD_OUTPUT}
+
+  # If an error occurs, run our error handler to output a tail of the build
+  trap 'error_handler' ERR
+
+  # Set up a repeating loop to send some output to Travis.
+  bash -c "while true; do echo \"\$(date) - building ...\"; sleep $PING_SLEEP; done" &
+  PING_LOOP_PID=$!
+
+  # Run integration tests, redirect output to tmp file
+  ./gradlew check -x test &> ${BUILD_OUTPUT}
+
+  # Kill ping process, output final
+  clean_up
+
+elif [[ "$1" == "unit" ]]; then
+  ./gradlew check -x integTest
+
+else
+  echo "ERROR: Invalid test type specified; must be either 'unit' or 'integration'"
+  exit 1
+fi
```

Comment:
Yep. Also, scalaStyle will produce a warning for files that don't end with newline.